### ISSUE TITLE
Enhancement - Add some polyline, polygon algorithms?

### ISSUE THREAD

**Issue by JTrotta** [Tronald/CoordinateSharp#56]
You call this library CoordinateSharp. You added some useful functions like distance between two points, and some astronomic calculations.
Why not add some useful algorithms, very often used with coordinates? I mean:

- If a point (lat, lon) in inside a polygon.
- How far is a point from a polyline.
- etc..
What do you think?

**Comment by JTrotta** [comment]
I could do it, if you like.

**Comment by Tronald** [comment]
That would be amazing! Thank you so much!

The only thing I would ask is that you build your polygone/polyline object models/logic in new class file for organizational purposes as it’s introducing a new scope to the library.

Anything you build for this library is greatly appreciated!

**Comment by JTrotta** [comment]
@Tronald , I suggest to create a new branch (develop?) where we can test new features and classes.
In the meanwhile I prepared the new class (not completed yet), just to check if you agree with code style.
Moreover do yo confirm you do not like external dependencies?
[GeoFence.zip](https://github.com/Tronald/CoordinateSharp/files/2260455/GeoFence.zip)

Another question: how do you create, in VS17, the multi target project? By hand (as I usually do), or can be done automatically? In the last case can you explain me how?

**Comment by Tronald** [comment]
Good idea. I created the branch [Feature_Develop](https://github.com/Tronald/CoordinateSharp/tree/Feature_Develop) for that purpose.

You are correct, I wish to avoid external dependencies. 

Your code style looks great, thank you for confirming.

Regarding multi-targeting in VS2017: This is much easier now in .NET standard, but still requires "hand jamming", but not as much. All I had to do was set the convert `<TargetFramework>` to the following.

```xml
 <TargetFrameworks>net40; netstandard1.3; netstandard1.4; netstandard2.0</TargetFrameworks>
```

It required some additional modifying to include the xml files and such. Have a look at [CoordinateSharp.csproj](https://github.com/Tronald/CoordinateSharp/blob/master/CoordinateSharp/CoordinateSharp.csproj). Now when I build the library it creates a library for each targeted framework and also creates a Nuget package automatically. Super convenient.
